debezium-core/src/test/java/io/debezium/generic/GenericConfigDefTest.java

```diff
@@ -0,0 +1,128 @@
+            try {
+                config = transformationClass.newInstance().config();
+            }
+            catch (InstantiationException | IllegalAccessException e) {
+                handleException(transformationClass, e);
+            }
+            if (null != config) {
+                checkConfig(transformationClass.getName(), config);
+            }
+        });
+
+        Set<Class<? extends org.apache.kafka.connect.transforms.predicates.Predicate>> predicateClasses = reflections
+                .getSubTypesOf(org.apache.kafka.connect.transforms.predicates.Predicate.class)
+                .stream().filter(FILTER_UNINSTATIONABLE_CLASSES)
+                .collect(Collectors.toSet());
+
+        LOGGER.info("Found PREDICATE classes: " + Arrays.toString(predicateClasses.toArray()));
+
+        predicateClasses.forEach(predicateClass -> {
+            LOGGER.info("Checking CONFIG_DEF for PREDICATES: " + predicateClass.getName() + " ...");
+            ConfigDef config = null;
+            try {
+                config = predicateClass.newInstance().config();
+            }
+            catch (InstantiationException | IllegalAccessException e) {
+                handleException(predicateClass, e);
+            }
+            if (null != config) {
+                checkConfig(predicateClass.getName(), config);
+            }
+        });
+    }
+
+    @Test
+    public void testConnectorConfigAllFieldsHaveDescription() {
+        Reflections reflections = new Reflections("io.debezium");
+        Set<Class<? extends CommonConnectorConfig>> classes = reflections.getSubTypesOf(CommonConnectorConfig.class)
+                .stream().filter(FILTER_UNINSTATIONABLE_CLASSES).collect(Collectors.toSet());
+        LOGGER.info("Found CONNECTOR CONFIG classes: " + Arrays.toString(classes.toArray()));
+
+        classes.forEach(connectorConfigClass -> {
+            LOGGER.info("Checking CONFIG_DEF for CONNECTOR_CONFIG: " + connectorConfigClass.getName() + " ...");
+            ConfigDef config = null;
+            try {
+                connectorConfigClass.getConstructor(Configuration.class).newInstance(Configuration.from(Collections.emptyMap()));
+            }
+            catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+                handleException(connectorConfigClass, e);
+                try {
+                    config = connectorConfigClass.getConstructor().newInstance().getConfigDef();
+                }
+                catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException | InstantiationException e2) {
+                    handleException(connectorConfigClass, e2);
+                }
+            }
+            if (null != config) {
+                checkConfig(connectorConfigClass.getName(), config);
+            }
+        });
+    }
+
+    private void handleException(Class<?> aClass, ReflectiveOperationException e) {
```

Comment:
Shouldn't it be actually thrown, so to not hide any unexpected errors?

Reply:
No, the handled exception happens when there's no empty default constructor or for `ConnectorConfig`'s when there's no empty default constructor or constructor that expects `Configuration` as parameter. This should only be logged, as this can happen to some weird SMT implementations, but it should not matter much actually. This is for example the case for the `io.debezium.junit.relational.TestRelationalDatabaseConfig` which has a weird constructor.

Reply:
Are there any other configs besides `TestRelationalDatabaseConfig`? If not, just add the default constructor to it, and it should address the issue?

Reply:
As I am loading all available implementations I will never know if there are new or changed implementations not compatible/instantiatable. Also a no-args default constructor might not easily be possible as the parent classes require config too and adding a constructor for a test case might also be a bad pattern.